sys/net/npf/npf_ruleset.c

```diff
@@ -716,6 +722,43 @@ npf_rule_setcode(npf_rule_t *rl, const int type, void *code, size_t size)
 	rl->r_jcode = npf_bpf_compile(code, size);
 }
 
+/* again, use a single rule getid function both uid and gids */
+static rid_t
+npf_rule_getrid(const nvlist_t *req, const char *name)
+{
```

Comment:
Why all this machinery? get dir of npf_rule_getrid and put all of it in npf_rule_setrid and operate on the pointer directly.